### PR TITLE
Fix the character field of RawKeyEvent is always null on iOS

### LINKED ISSUE #133956: [Widget Tests] - Key character resolution is broken on iOS
In widget tests, when simulating key presses that correspond to characters, the iOS character resolution is broken.

I'll illustrate this with two tests - one that shows the issues with iOS, and one that shows no such issues with Android.

Bug Repro:

```dart
testWidgets("bug repro", (widgetTester) async {
      debugDefaultTargetPlatformOverride = TargetPlatform.linux;

      await widgetTester.pumpWidget(
        MaterialApp(
          home: Scaffold(
            body: Focus(
              onKey: (FocusNode node, RawKeyEvent keyEvent) {
                print("Key event");
                print("$keyEvent");
                print("Character ${keyEvent.character}");
                print("");

                return KeyEventResult.ignored;
              },
              autofocus: true,
              child: const SizedBox.expand(),
            ),
          ),
        ),
      );

      await widgetTester.sendKeyDownEvent(LogicalKeyboardKey.shift, platform: "ios");

      await widgetTester.sendKeyDownEvent(LogicalKeyboardKey.keyG, platform: "ios", character: "G");
      await widgetTester.sendKeyUpEvent(LogicalKeyboardKey.keyG, platform: "ios");

      await widgetTester.sendKeyUpEvent(LogicalKeyboardKey.shift, platform: "ios");

      debugDefaultTargetPlatformOverride = null;
    });
```

Bug repro output - notice that there's never a "G" character reported:

```
Key event
RawKeyDownEvent#3b0d6(logicalKey: LogicalKeyboardKey#df100(keyId: "0x200000102", keyLabel: "Shift Left", debugName: "Shift Left"), physicalKey: PhysicalKeyboardKey#ed430(usbHidUsage: "0x000700e1", debugName: "Shift Left"), repeat: false)
Character null

Key event
RawKeyDownEvent#55276(logicalKey: LogicalKeyboardKey#c9177(keyId: "0x00000047", keyLabel: "G", debugName: "Key G"), physicalKey: PhysicalKeyboardKey#8c52b(usbHidUsage: "0x0007000a", debugName: "Key G"), repeat: false)
Character null

Key event
RawKeyUpEvent#a5680(logicalKey: LogicalKeyboardKey#23b97(keyId: "0x00000067", keyLabel: "G", debugName: "Key G"), physicalKey: PhysicalKeyboardKey#8c52b(usbHidUsage: "0x0007000a", debugName: "Key G"))
Character null

Key event
RawKeyUpEvent#8f6fc(logicalKey: LogicalKeyboardKey#df100(keyId: "0x200000102", keyLabel: "Shift Left", debugName: "Shift Left"), physicalKey: PhysicalKeyboardKey#ed430(usbHidUsage: "0x000700e1", debugName: "Shift Left"))
Character null
```

Here's the same test except using "android" as the simulated platform:

```dart
testWidgets("bug counter example", (widgetTester) async {
      debugDefaultTargetPlatformOverride = TargetPlatform.linux;

      await widgetTester.pumpWidget(
        MaterialApp(
          home: Scaffold(
            body: Focus(
              onKey: (FocusNode node, RawKeyEvent keyEvent) {
                print("Key event");
                print("$keyEvent");
                print("Character ${keyEvent.character}");
                print("");

                return KeyEventResult.ignored;
              },
              autofocus: true,
              child: const SizedBox.expand(),
            ),
          ),
        ),
      );

      await widgetTester.sendKeyDownEvent(LogicalKeyboardKey.shift, platform: "android");

      await widgetTester.sendKeyDownEvent(LogicalKeyboardKey.keyG, platform: "android", character: "G");
      await widgetTester.sendKeyUpEvent(LogicalKeyboardKey.keyG, platform: "android");

      await widgetTester.sendKeyUpEvent(LogicalKeyboardKey.shift, platform: "android");

      debugDefaultTargetPlatformOverride = null;
    });
```

Here's the output that shows things work as expected on Android:

```
Key event
RawKeyDownEvent#fcd4b(logicalKey: LogicalKeyboardKey#df100(keyId: "0x200000102", keyLabel: "Shift Left", debugName: "Shift Left"), physicalKey: PhysicalKeyboardKey#ed430(usbHidUsage: "0x000700e1", debugName: "Shift Left"), repeat: false)
Character null

Key event
RawKeyDownEvent#e570e(logicalKey: LogicalKeyboardKey#23b97(keyId: "0x00000067", keyLabel: "G", debugName: "Key G"), physicalKey: PhysicalKeyboardKey#8c52b(usbHidUsage: "0x0007000a", debugName: "Key G"), repeat: false)
Character G

Key event
RawKeyUpEvent#2a444(logicalKey: LogicalKeyboardKey#23b97(keyId: "0x00000067", keyLabel: "G", debugName: "Key G"), physicalKey: PhysicalKeyboardKey#8c52b(usbHidUsage: "0x0007000a", debugName: "Key G"))
Character null

Key event
RawKeyUpEvent#c6b9c(logicalKey: LogicalKeyboardKey#df100(keyId: "0x200000102", keyLabel: "Shift Left", debugName: "Shift Left"), physicalKey: PhysicalKeyboardKey#ed430(usbHidUsage: "0x000700e1", debugName: "Shift Left"))
Character null
```

### PR DESCRIPTION
**Pull request by LinXunFeng**
Fixes https://github.com/flutter/flutter/issues/133956

### CODE REVIEW

File: packages/flutter/lib/src/services/raw_keyboard.dart [hunk 1]
```diff
@@ -356,6 +356,10 @@ abstract class RawKeyEvent with Diagnosticable {
             keyCode: message['keyCode'] as int? ?? 0,
             modifiers: message['modifiers'] as int? ?? 0,
           );
+          final Object? characters = message['characters'];
```
Comment (dkwingsmt) [inline packages/flutter/lib/src/services/raw_keyboard.dart:359]: Sorry but how is this line different from L354?

**Review by dkwingsmt — APPROVED** [review]
LGTM

**Comment by goderbauer** [comment]
(triage): @LinXunFeng The google testing check requires that this branch is rebased in order to run. Sadly, github doesn't allow me to rebase it from my end. Can you please rebase it to the latest master so that we can merge it?

**Review by cbracken — APPROVED** [review]
LGTM stamp from a Japanese personal seal

**Comment by LinXunFeng** [comment]
@goderbauer done, the google testing passed.

**Pull request merged**